payroll/models/hr_salary_rule.py

```diff
@@ -274,8 +266,7 @@ def _satisfy_condition(self, localdict):
                 )
         else:  # python code
             try:
-                safe_eval(self.condition_python, localdict, mode="exec", nocopy=True)
-                return "result" in localdict and localdict["result"] or False
+                return self._satisfy_condition_python(localdict)
```

Comment:
We can do in the same suggested in my last comment with _satisfy_condition function.